packages/playwright-ct/src/fixture.ts

```diff
@@ -0,0 +1,615 @@
+    if (!id) {
+      return;
+    }
+
+    try {
+      await page.evaluate(ReplayAddAnnotation, [
+        event,
+        id,
+        JSON.stringify({ ...detail, test: testData }),
+      ] as const);
+    } catch (error) {
+      logError("ReplayCTFixture:FailedToAddAnnotation", { error });
+    }
+  }
+
+  // Set up client instrumentation listener
+  const csiListener: ClientInstrumentationListener = {
+    onApiCallBegin: ({ userData, params, apiName, frames }) => {
+      const step: TestStepInternal | undefined = userData;
+
+      if (!step?.stepId) {
+        return;
+      }
+
+      if (isReplayAnnotation(params)) {
+        ignoredSteps.add(step.stepId);
+        return;
+      }
+
+      if (expectSteps.has(step.stepId)) {
+        return;
+      }
+
+      if (!frames.length) {
+        ignoredSteps.add(step.stepId);
+        return;
+      }
+
+      testInfo.attach(`replay:step:start`, {
+        body: JSON.stringify({
+          event: "step:start",
+          id: step.stepId,
+          test: testData,
+          apiName,
+          category: step.category,
+          frames,
+          params: params ?? {},
+          title: step.title,
+          hook: testInfo._currentHookType?.(),
+        }),
+        contentType: REPLAY_CONTENT_TYPE,
+      });
+
+      return addAnnotation("step:start", step.stepId, {
+        apiName,
+        category: step.category,
+        title: step.title,
+        params: params || {},
+        frames,
+        hook: testInfo._currentHookType?.(),
+      });
+    },
+
+    onApiCallEnd: ({ userData, error, params }) => {
+      const step: TestStepInternal | undefined = userData;
+      if (!step?.stepId || ignoredSteps.has(step.stepId)) {
+        return;
+      }
+
+      testInfo.attach(`replay:step:end`, {
+        body: JSON.stringify({
+          event: "step:end",
+          id: step.stepId,
+          test: testData,
+          error: error ? parseError(error) : null,
+        }),
+        contentType: REPLAY_CONTENT_TYPE,
+      });
+
+      return addAnnotation("step:end", step.stepId, {
+        error: error ? parseError(error) : null,
+      });
+    },
+  };
+
+  const clientInstrumentation = playwright._instrumentation;
+  clientInstrumentation.addListener(csiListener);
+
+  // Enhanced mount function that tracks component lifecycle
+  const enhancedMount = async (component: any, options?: any) => {
+    console.log(
+      "[replay.io CT]: Enhanced mount called for component:",
+      component?.type?.name || component?.type || "Component"
+    );
+    const componentId = uuid();
+    const componentName = component?.type?.name || component?.type || "Component";
+
+    // Track mount start
+    const mountStepId = uuid();
+    testInfo.attach(`replay:step:start`, {
+      body: JSON.stringify({
+        event: "step:start",
+        id: mountStepId,
+        test: testData,
+        apiName: "mount",
+        category: "pw:api",
+        title: `mount: ${componentName}`,
```

Comment:
this feels like a hack (it's not a `pw:api` per-se ;p) - how do we use those attachments? do we need them?